libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,203 @@
+        following = true;
+      } else if (not arg.starts_with("-")) {
+        std::error_code err{};
+        auto status = std::filesystem::status(arg, err);
+        if (err) {
+          return caf::make_error(ec::parse_error,
+                                 fmt::format("could not access file {}: {}",
+                                             arg, err));
+        }
+        is_socket = (status.type() == std::filesystem::file_type::socket);
+        path = arg;
+      } else {
+        return caf::make_error(
+          ec::invalid_argument,
+          fmt::format("unexpected argument for 'file' connector: {}", arg));
+      }
+    }
+    if (path.empty()) {
+      return caf::make_error(ec::syntax_error,
+                             fmt::format("no file specified"));
+    }
+    auto fd = STDIN_FILENO;
+    if (is_socket) {
+      if (path == stdin_path) {
+        return caf::make_error(ec::filesystem_error, "cannot use STDIN as UNIX "
+                                                     "domain socket");
+      }
+      auto uds = detail::unix_domain_socket::connect(path);
+      if (!uds) {
+        return caf::make_error(
+          ec::filesystem_error,
+          fmt::format("Unable to connect to UNIX domain socket at {}:", path));
+      }
+      fd = uds.fd;
+    } else {
+      if (path != stdin_path) {
+        fd = ::open(path.c_str(), std::ios_base::binary | std::ios_base::in);
+        if (fd == -1) {
+          return caf::make_error(ec::filesystem_error,
+                                 fmt::format("open(2) for file {} failed {}:",
+                                             path, std::strerror(errno)));
+        }
+      }
+    }
+    return std::invoke(
+      [](auto timeout, auto fd, auto following) -> generator<chunk_ptr> {
+        auto in_buf = detail::fdinbuf(fd, max_chunk_size);
+        in_buf.read_timeout() = timeout;
+        auto current_data = std::vector<std::byte>{};
+        current_data.reserve(max_chunk_size);
+        auto eof_reached = false;
+        while (following or not eof_reached) {
+          auto current_char = in_buf.sbumpc();
+          if (current_char != detail::fdinbuf::traits_type::eof()) {
+            current_data.emplace_back(static_cast<std::byte>(current_char));
+          }
+          if (current_char == detail::fdinbuf::traits_type::eof()
+              or current_data.size() == max_chunk_size) {
+            eof_reached = (current_char == detail::fdinbuf::traits_type::eof()
+                           and not in_buf.timed_out());
+            if (eof_reached and current_data.empty() and not following) {
+              break;
+            }
+            auto chunk = chunk::make(std::exchange(current_data, {}));
+            co_yield std::move(chunk);
+            if (eof_reached and not following) {
+              break;
+            }
+            if (not eof_reached or following) {
+              current_data.reserve(max_chunk_size);
+            }
```

Comment:
```suggestion
            current_data.reserve(max_chunk_size);
```
This condition is the negation of the previous one, and therefore always true.

Reply:
Removed condition in https://github.com/tenzir/vast/pull/3088/commits/26c17b32e7ba53a27f5027a6ab918f74b40d215c